back/app/services/form_logic_service.rb

```diff
@@ -38,6 +38,16 @@ def valid?
     end
   end
 
+  def remove_select_logic_option_from_custom_fields(frozen_custom_field_option)
+    custom_field = fields.first
```

Comment:
I think it's a little bit risky to assume that the code that calls the service will have initialized it with the corresponding field as the first field in the list. Perhaps we could get the custom field from the option instead?